### ISSUE TITLE
Thresholds not working properly in Safari/iOS

### ISSUE THREAD

**Issue by RubenKos** [w3c/IntersectionObserver#307]
Hi all,

Looking forward to using this new API in my projects as it seem to improve performance a lot!

Unfortunately, after running a few tests and prototypes I noticed that the Thresholds are not supporting properly in the Polyfill for Safari and iOS.

I have taken the [Thresholds example](https://developer.mozilla.org/en-US/docs/Web/API/Intersection_Observer_API#Thresholds) from the Moz Docs and put it in a Pen, where I also load in the Polyfill. 

This means that the example works cross-browsers, but you will notice that the frequency of callback is far fewer in Safari (11.1, macOS) and iOS (11.4) than in other browsers resulting in different results cross-browser unfortunately. 

The Pen can be viewed [right here](https://codepen.io/senza12/pen/QxvqmM).

Appreciate any help, hopefully this can be solved so IntersectionObserver becomes a great alternative to all the expensive libraries out there.

Thank you in advance.

**Comment by simeyla** [comment]
The bigger problem with iOS Safari is that if you don't lock the menu and toolbar then the viewport size changes as the toolbars auto-hide themselves. This leads to items that are visible responding with 0%.

I can't immediately see what might be wrong with your pen - on iOS for me it always shows 100% - but that could be because it's in an iframe.

I was wondering if your declaration of observerOptions outside the loop was an issue if it were being reused and not cloned internally in the IntersectionObserver.

I didn't see the problem you mention - in fact I found it to work very speedily indeed on iOS to respond to thresholds almost instantly as I passed them (I had intervals at 10% for testing).

Can you explain what frequency you see on iOS (since I only see 100% for everything on my iPhone via the pen).

**Comment by RubenKos** [comment]
I noticed the problem regarding the menu and toolbar and there are already many issues posted in this repository around this, I agree this is also an issue and can hopefully be fixed soon.

I didn't test my Pen on iOS before posting and it does indeed not work, most likely because of the iframe. It is probably best to test in Safari as it behaves the same as on iOS when I tested my prototypes locally.

I took the initial code from the Mozilla documentation so I am assuming the declaration of observerOptions outside the loop is not an issue but could be wrong there.

The 10% intervals are working okay, but when looking at smaller intervals of 1% you see a clear difference between the behaviours of let's say Chrome and Safari. In Chrome it updates every 1% were in Safari it will skip a few randomly and doesn't behave as smooth at all.

Hope that clarifies things a bit.
